### PR TITLE
docs: Cover exception changes in the migration guide

### PR DESCRIPTION
**Pull request by dazuma**
@quartzmo How does this section on exceptions look? After we get this the way we want it, I'll replicate it across the rest of the migration guides.

### CODE REVIEW

File: google-cloud-asset/MIGRATING.md [hunk 1]
```diff
@@ -29,6 +29,10 @@ To summarize:
     resource paths. These paths are now instance methods on the client objects,
     and are also available in a separate paths module. See
     [Resource Path Helpers](#resource-path-helpers) for more info.
+ *  Previously, RPC exceptions were subclasses of `Google::Gax::GaxError`,
+    defined in the `google-gax` gem. Now, RPSs throw exceptions subclassing
+    `Google::Cloud::Error`, defined in the `google-cloud-errors` gem. See
+    [Error Handling](#error-handling) for more info.
```
Comment (quartzmo) [inline google-cloud-asset/MIGRATING.md:35]: [Handling Errors](#handling-errors)

File: google-cloud-asset/MIGRATING.md [hunk 2]
```diff
@@ -29,6 +29,10 @@ To summarize:
     resource paths. These paths are now instance methods on the client objects,
     and are also available in a separate paths module. See
     [Resource Path Helpers](#resource-path-helpers) for more info.
+ *  Previously, RPC exceptions were subclasses of `Google::Gax::GaxError`,
+    defined in the `google-gax` gem. Now, RPSs throw exceptions subclassing
```
Comment (quartzmo) [inline google-cloud-asset/MIGRATING.md:33]: Now that the client no longer uses `google-gax`, RPCs throw...

File: google-cloud-asset/MIGRATING.md [hunk 3]
```diff
@@ -269,6 +273,47 @@ def foo
 end
 ```
 
+### Handling Errors
+
+The client reports standard HTTP and gRPC error codes by raising exceptions.
```
Comment (quartzmo) [inline google-cloud-asset/MIGRATING.md:278]: Can you link to gRPC error codes?

Comment (dazuma) [reply google-cloud-asset/MIGRATING.md]: Done

File: google-cloud-asset/MIGRATING.md (hunk 2 above)
Comment (dazuma) [reply google-cloud-asset/MIGRATING.md]: Updated to remove the references to the gems (which is probably not necessary for the summary.)

File: google-cloud-asset/MIGRATING.md (hunk 1 above)
Comment (dazuma) [reply google-cloud-asset/MIGRATING.md]: Whups, good catch!

**Pull request merged**